### ISSUE TITLE
GenerateBindingRedirects is always out of date, breaks parallel builds

### ISSUE THREAD

**Issue by aaronla-ms** [dotnet/msbuild#6309]
### Issue Description

GenerateBindingRedirects in Microsoft.Common.CurrentVersion.targets is always out of date. This causes problems in parallel builds on the same machine.

GenerateBindingRedirects in Microsoft.Common.CurrentVersion.targets is defined as

```xml
<Target Name="GenerateBindingRedirects"
    Inputs="$(MSBuildAllProjects);@(AppConfigFile);$(ResolveAssemblyReferencesStateFile);$(IntermediateOutputPath);@(SuggestedBindingRedirects)"
 Outputs="$(_GenerateBindingRedirectsIntermediateAppConfig)"
 Condition="'$(AutoGenerateBindingRedirects)' == 'true' and '$(GenerateBindingRedirectsOutputType)' == 'true'">
```

Notice `@(SuggestedBindingRedirects)` in the input list. This contains items such as "System.Fabric, Culture=neutral, PublicKeyToken=31bf3856ad364e35", which is not an actual file. As a consequence, this target is always out of date. This in turn causes writes to the .exe.config file.

This is problematic in parallel builds. GenerateBindingRedirects gets run as part of ResolveReferences and Publish, which sfproj projects downstream will use to package up the service for deployment in a service fabric application. When two sfproj's are downstream, they will both invoke Publish at the same time.

While  isn't a problem on parallel _msbuild_ --  GenerateBindingRedirects will get 'deduplicated' and likely only run once per project -- it _is_ a problem on other parallel build engines, such as our internal CloudBuild infrastructure. Cloudbuild will invoke the sfproj's with `msbuild.exe /p:BuildProjectReferences=false`, expecting that to cause upstream projects to have no side effects, perform no writes. However, GenerateBindingRedirects will still write to the .exe.config file even when BuildProjectReferences=false. This occasionally causes builds to fail when racing on accessing the .exe.config file.

### Steps to Reproduce

See repro.cmd in linked file (Microsoft users only):
https://microsoft-my.sharepoint.com/:u:/p/aaronla_ntdev/ESzNWwBjXvFOqMYHMx68Z7sBvuk_ZitNfJFLDOWGV4GXcQ?e=cnKIT0

### Expected Behavior

During build of application1, no writes to obj\Debug\net462\win7-x64\Stateless1.exe.config observed

### Actual Behavior

obj\Debug\net462\win7-x64\Stateless1.exe.config is written to, potentially breaking parallel builds. The logs indicate this is because:

  Input file "System.Fabric, Culture=neutral, PublicKeyToken=31bf3856ad364e35" does not exist.

### Analysis

GenerateBindingRedirects target has a bad entry in Inputs list. The `@(SuggestedBindingRedirects)` is not files, so should not be on the Inputs list.

### Versions & Configurations

Reproduce under Visual Studio 2019 and Visual Studio 2019 Preview. Repro above run with msbuild  16.9.0+57a23d249

### Attach a binlog

Can be provided on request to a storage location suitable for customer data. Not comfortable posting binlog publicly.

**Comment by rainersigwald** [comment]
Duplicate of #5943
